syntax = "proto3";

package leads;

service Classifier {
  rpc Classify (ClassificationRequest) returns (ClassificationResponse) {}
}

// Represents a social media message.
message ClassificationRequest {
  string title = 1;
  string body = 2;
  repeated string topics = 3;
}

// The response message containing the classification result.
message ClassificationResponse {
  repeated Topic matched_topics = 1;
  float score_threshold = 2;
  float execution_time = 3;
}

message Topic {
   string label = 1;
   float score = 2;
}
